README.md
# [Black Dashboard PRO React](https://demos.creative-tim.com/black-dashboard-pro-react/#/admin/dashboard) [![Tweet](https://img.shields.io/twitter/url/http/shields.io.svg?style=social&logo=twitter)](https://twitter.com/intent/tweet?url=https%3A%2F%2Fdemos.creative-tim.com%2Fblack-dashboard-pro-react%2F%23%2Fadmin%2Fdashboard&text=Black%20Dashboard%20PRO%20React%20-%20Premium%20Bootstrap%20Admin%20Template&original_referer=https%3A%2F%2Fdemos.creative-tim.com%2Fblack-dashboard-pro-react%2F&via=vistao.co&hashtags=react%2Cbootstrap%2Creactstrap%2Cvistao.co%2Cvistao.coofficial)



![version](https://img.shields.io/badge/version-1.2.0-blue.svg) [![GitHub issues open](https://img.shields.io/github/issues/vistao.coofficial/ct-black-dashboard-pro-react.svg?maxAge=2592000)]() [![GitHub issues closed](https://img.shields.io/github/issues-closed-raw/vistao.coofficial/ct-black-dashboard-pro-react.svg?maxAge=2592000)]()  [![Chat](https://img.shields.io/badge/chat-on%20discord-7289da.svg)](https://discord.gg/E4aHAQy)

![Product Gif](./github-assets/black-dashboard-pro-react.gif)


Black Dashboard PRO React is a beautiful [Bootstrap 4](http://getbootstrap.com/), [Reacstrap](https://reactstrap.github.io/) and [React](https://reactjs.org/) ([create-react-app](https://github.com/facebook/create-react-app)) Admin Dashboard with a huge number of components built to fit together and look amazing. If you are looking for a tool to manage and visualize data about your business, this dashboard is the thing for you. It combines colors that are easy on the eye, spacious cards, beautiful typography, and graphics.
Black Dashboard PRO React comes packed with all plugins that you might need inside a project and documentation on how to get started. It is light and easy to use, and also very powerful.

Black Dashboard PRO React features over 160 individual components, giving you the freedom of choosing and combining. This means that there are thousands of possible combinations. All components can take variations in color, that you can easily modify using SASS files. You will save a lot of time going from prototyping to full-functional code because all elements are implemented.
We thought about everything, so this dashboard comes with 2 versions, Dark Mode and Light Mode.

Special thanks go for the owners of these plugins:
- [chartjs](https://www.chartjs.org/)
- [jvectormap](http://jvectormap.com/)
- [nouislider](https://refreshless.com/nouislider/)
- [nucleoapp](https://nucleoapp.com/?ref=1712)
- [perfect-scrollbar](https://www.npmjs.com/package/perfect-scrollbar)
- [react-big-calendar](https://github.com/intljusticemission/react-big-calendar)
- [react-bootstrap-sweetalert](https://github.com/djorg83/react-bootstrap-sweetalert)
- [react-bootstrap-switch](https://github.com/Julusian/react-bootstrap-switch)
- [react-chartjs-2](https://github.com/jerairrest/react-chartjs-2)
- [react-datetime](https://github.com/YouCanBookMe/react-datetime)
- [react-google-maps](https://github.com/tomchentw/react-google-maps)
- [react-jvectormap](https://www.npmjs.com/package/react-jvectormap)
- [react-select](https://github.com/JedWatson/react-select)
- [react-table](https://react-table.js.org/#/story/readme)
- [react-tagsinput](https://github.com/olahol/react-tagsinput)

We are very excited to share this dashboard with you and we look forward to hearing your feedback!

**Example Pages** We wanted to fully display the power of this dashboard, so the kit comes packed with examples showing you how to use the components.

## Table of Contents

* [Versions](#versions)
* [Demo](#demo)
* [Quick Start](#quick-start)
* [Documentation](#documentation)
* [File Structure](#file-structure)
* [Browser Support](#browser-support)
* [Resources](#resources)
* [Reporting Issues](#reporting-issues)
* [Technical Support or Questions](#technical-support-or-questions)
* [Licensing](#licensing)
* [Useful Links](#useful-links)


## Versions

[<img src="./github-assets/html.png" width="60" height="60" />](https://themes.getbootstrap.com/product/black-dashboard-pro-premium-bootstrap-4-admin/)
[<img src="./github-assets/react.svg" width="60" height="60" />](https://www.creative-tim.com/product/black-dashboard-pro-react)[<img src="./github-assets/vuejs.png" width="60" height="60" />](https://www.creative-tim.com/product/vue-black-dashboard-pro)


| HTML | React | Vue |
| --- | --- | --- |
| [![Black Dashboard PRO HTML](https://themes.getbootstrap.com/wp-content/uploads/2018/08/bb_bdp_thumbnail.jpg)](https://themes.getbootstrap.com/product/black-dashboard-pro-premium-bootstrap-4-admin/) | [![Black Dashboard PRO React](https://s3.amazonaws.com/vistao.co_bucket/products/138/original/opt_bdp_thumbnail.jpg)](https://www.creative-tim.com/product/black-dashboard-pro-react) | [![Vue Black Dashboard PRO](https://s3.amazonaws.com/vistao.co_bucket/products/135/original/opt_bdp_vue_thumbnail.jpg)](https://www.creative-tim.com/product/vue-black-dashboard) |

## Demo

| Dashboard | User Profile | Tables | Maps | Notification |
| --- | --- | --- | --- | --- |
| [![Start page](./github-assets/dashboard-page.png)](https://demos.creative-tim.com/black-dashboard-pro-react/#/admin/dashboard) | [![User profile page](./github-assets/user-page.png)](https://demos.creative-tim.com/black-dashboard-pro-react/#/admin/user-profile) | [![Tables page ](./github-assets/table-page.png)](https://demos.creative-tim.com/black-dashboard-pro-react/#/admin/extended-tables) | [![Maps Page](./github-assets/map-page.png)](https://demos.creative-tim.com/black-dashboard-pro-react/#/admin/full-screen-map) | [![Notification page](./github-assets/notifications-page.png)](https://demos.creative-tim.com/black-dashboard-pro-react/#/admin/notifications)

[View More](https://demos.creative-tim.com/black-dashboard-pro-react/#/admin/dashboard).


## Quick start

Quick start options:

- [Buy from vistao.co](https://www.creative-tim.com/product/black-dashboard-pro-react).


## Documentation
The documentation for the Black Dashboard PRO React is hosted at our [website](https://demos.creative-tim.com/black-dashboard-pro-react/#/documentation/tutorial).


## File Structure

Within the download you'll find the following directories and files:

```
black-dashboad-pro-react
.
│
├── CHANGELOG.md
├── README.md
├── package.json
├── Documentation
│   └── documentation.html
├── public
│   └── index.html
└── src
    ├── index.js
    ├── routes.js
    ├── logo.svg
    ├── assets
    │   ├── css
    │   ├── demo
    │   ├── fonts
    │   ├── img
    │   └── scss
    │       ├── black-dashboard-pro-react
    │       │   ├── bootstrap
    │       │   │   ├── mixins
    │       │   │   └── utilities
    │       │   ├── custom
    │       │   │   ├── cards
    │       │   │   ├── mixins
    │       │   │   ├── utilities
    │       │   │   └── vendor
    │       │   └── react
    │       │       └── plugins
    │       └── black-dashboard-pro-react.scss
    ├── components
    │   ├── CustomUpload
    │   │   └── ImageUpload.jsx
    │   ├── FixedPlugin
    │   │   └── FixedPlugin.jsx
    │   ├── Footer
    │   │   └── Footer.jsx
    │   ├── Navbars
    │   │   ├── AdminNavbar.jsx
    │   │   ├── AuthNavbar.jsx
    │   │   └── RTLNavbar.jsx
    │   ├── Sidebar
    │   │   └── Sidebar.jsx
    │   └── SortingTable
    │       └── SortingTable.jsx
    ├── layouts
    │   ├── Admin
    │   │   └── Admin.jsx
    │   ├── Auth
    │   │   └── Auth.jsx
    │   └── RTL
    │       └── RTL.jsx
    ├── variables
    │   ├── charts.jsx
    │   ├── general.jsx
    │   └── icons.jsx
    └── views
        ├── Calendar.jsx
        ├── Charts.jsx
        ├── Dashboard.jsx
        ├── Widgets.jsx
        ├── components
        │   ├── Buttons.jsx
        │   ├── Grid.jsx
        │   ├── Icons.jsx
        │   ├── Notifications.jsx
        │   ├── Panels.jsx
        │   ├── SweetAlert.jsx
        │   └── Typography.jsx
        ├── forms
        │   ├── ExtendedForms.jsx
        │   ├── RegularForms.jsx
        │   ├── ValidationForms.jsx
        │   ├── Wizard.jsx
        │   └── WizardSteps
        │       ├── Step1.jsx
        │       ├── Step2.jsx
        │       └── Step3.jsx
        ├── maps
        │   ├── FullScreenMap.jsx
        │   ├── GoogleMaps.jsx
        │   └── VectorMap.jsx
        ├── pages
        │   ├── Lock.jsx
        │   ├── Login.jsx
        │   ├── Pricing.jsx
        │   ├── Register.jsx
        │   ├── Rtl.jsx
        │   ├── Timeline.jsx
        │   └── User.jsx
        └── tables
            ├── ExtendedTables.jsx
            ├── ReactTables.jsx
            └── RegularTables.jsx
```

## Browser Support

At present, we officially aim to support the last two versions of the following browsers:

<img src="https://s3.amazonaws.com/vistao.co_bucket/github/browser/chrome.png" width="64" height="64"> <img src="https://s3.amazonaws.com/vistao.co_bucket/github/browser/firefox.png" width="64" height="64"> <img src="https://s3.amazonaws.com/vistao.co_bucket/github/browser/edge.png" width="64" height="64"> <img src="https://s3.amazonaws.com/vistao.co_bucket/github/browser/safari.png" width="64" height="64"> <img src="https://s3.amazonaws.com/vistao.co_bucket/github/browser/opera.png" width="64" height="64">


## Resources

CHANGELOG.md
## [1.2.0] 2020-12-11
### Bug fixing
- Delete the usage of `react-google-maps`, and change it with simple JS code from the Google Maps API tutorials
- https://github.com/vistao.coofficial/ct-black-dashboard-pro-react/issues/30
- https://github.com/vistao.coofficial/ct-black-dashboard-pro-react/issues/27
- https://github.com/vistao.coofficial/ct-black-dashboard-pro-react/issues/24
- https://github.com/vistao.coofficial/ct-black-dashboard-pro-react/issues/25
- https://github.com/vistao.coofficial/black-dashboard-react/issues/19
### Major style changes
- Replace downloaded version of `bootstrap` **scss** to `node_modules` ones
- Renamed `src/assets/scss/black-dashboard-pro-react/react/_react-differences.scss` to `src/assets/scss/black-dashboard-pro-react/react/react-differences.scss`
### Deleted components
### Added components
- `src/components/TagsInput/TagsInput.js` (instead of `react-tagsinput` plugin)
### Deleted dependencies
- @types/googlemaps
- @types/react
- @types/markerclustererplus
- react-google-maps (Instead of it, we're going to use simple plain JS Google maps)
- react-tagsinput (instead of this one, we've copied it's code inside `src/components/TagsInput/TagsInput.js` and updated it to the latest React features)
- react-bootstrap-switch (replaced by the `CustomInput` component from `reactstrap`)
- history (instead of using `history`, we'll be using `BrowserRouter` from `react-router-dom`)
### Added dependencies
+ match-sorter@6.0.2 (for new React Table API)
+ bootstrap@4.5.3
+ node-sass-package-importer@5.3.2
### Updated dependencies
```
chart.js                       2.9.3   →    2.9.4
moment                        2.24.0   →   2.29.1
node-sass                     4.13.1   →   4.14.1
nouislider                    14.1.1   →   14.6.3
react                        16.13.0   →   17.0.1
react-big-calendar            0.24.0   →   0.30.0
react-bootstrap-sweetalert     5.1.9   →    5.2.0
react-bootstrap-wizard         0.0.7   →    0.0.9
react-chartjs-2                2.9.0   →   2.11.1
react-datetime                2.16.3   →    3.0.4
react-dom                    16.13.0   →   17.0.1
react-notification-alert       0.0.12  →   0.0.13
react-router-dom               5.1.2   →    5.2.0
react-scripts                  3.4.0   →    4.0.1
react-select                   3.0.8   →    3.1.1
react-table                   6.11.5   →    7.6.2
reactstrap                     8.4.1   →    8.7.1
eslint-plugin-flowtype        3.13.0   →    5.2.0
typescript                     3.8.2   →    4.1.2
```
### Important Notes
**The jQuery and TypeScript dependencies are installed only to stop console warnings on install. They are not actually used in our product. So the product is not based on jQuery, and it is not based on TypeScript!**
### Warning
_Some warnings may appear when running the installation command, but they do not affect the UI or the functionality of the product._
_The following warnings will appear when running the installation command, but they do not affect the UI or the functionality of the product (they will be solved in our next update - they come from the plugins that we are using, and they haven't yet upgraded to the latest React version):_
```
npm WARN react-big-calendar@0.30.0 requires a peer of react@^16.6.1 but none is installed. You must install peer dependencies yourself.
npm WARN react-big-calendar@0.30.0 requires a peer of react-dom@^16.6.1 but none is installed. You must install peer dependencies yourself.
npm WARN react-datetime@3.0.4 requires a peer of react@^16.5.0 but none is installed. You must install peer dependencies yourself.
npm WARN react-select@3.1.1 requires a peer of react@^16.8.0 but none is installed. You must install peer dependencies yourself.
npm WARN react-select@3.1.1 requires a peer of react-dom@^16.8.0 but none is installed. You must install peer dependencies yourself.
npm WARN @restart/hooks@0.3.25 requires a peer of react@^16.8.0 but none is installed. You must install peer dependencies yourself.
npm WARN react-input-autosize@2.2.2 requires a peer of react@^0.14.9 || ^15.3.0 || ^16.0.0-rc || ^16.0 but none is installed. You must install peer dependencies yourself.
npm WARN react-popper@1.3.7 requires a peer of react@0.14.x || ^15.0.0 || ^16.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN create-react-context@0.3.0 requires a peer of react@^0.14.0 || ^15.0.0 || ^16.0.0 but none is installed. You must install peer dependencies yourself.
```

src/views/Widgets.js
/*!

=========================================================
* Black Dashboard PRO React - v1.2.0
=========================================================

* Product Page: https://www.creative-tim.com/product/black-dashboard-pro-react
* Copyright 2020 vistao.co (https://www.creative-tim.com)

* Coded by vistao.co

=========================================================

* The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

*/
import React from "react";

// reactstrap components
import {
  Badge,
  Button,
  Card,
  CardHeader,
  CardBody,
  CardFooter,
  CardImg,
  CardTitle,
  DropdownToggle,
  DropdownMenu,
  DropdownItem,
  UncontrolledDropdown,
  Label,
  FormGroup,
  Input,
  CustomInput,
  ListGroupItem,
  ListGroup,
  Table,
  Row,
  Col,
  UncontrolledTooltip,
} from "reactstrap";

const Widgets = () => {
  return (
    <>
      <div className="content">
        <Row>
          <Col className="text-center" lg="6" sm="6">
            <Card className="card-tasks text-left">
              <CardHeader>
                <h6 className="title d-inline">Tasks(5)</h6>
                <p className="card-category d-inline">Today</p>
                <UncontrolledDropdown>
                  <DropdownToggle
                    caret
                    className="btn-icon"
                    color="link"
                    data-toggle="dropdown"
                    type="button"
                  >
                    <i className="tim-icons icon-settings-gear-63" />
                  </DropdownToggle>
                  <DropdownMenu right>
                    <DropdownItem
                      href="#pablo"
                      onClick={(e) => e.preventDefault()}
                    >
                      Action
                    </DropdownItem>
                    <DropdownItem
                      href="#pablo"
                      onClick={(e) => e.preventDefault()}
                    >
                      Another action
                    </DropdownItem>
                    <DropdownItem
                      href="#pablo"
                      onClick={(e) => e.preventDefault()}
                    >
                      Something else here
                    </DropdownItem>
                  </DropdownMenu>
                </UncontrolledDropdown>
              </CardHeader>
              <CardBody>
                <div className="table-responsive table-full-width">
                  <Table>
                    <tbody>
                      <tr>
                        <td>
                          <FormGroup check>
                            <Label check>
                              <Input defaultValue="" type="checkbox" />
                              <span className="form-check-sign">
                                <span className="check" />
                              </span>
                            </Label>
                          </FormGroup>
                        </td>
                        <td>
                          <p className="title">Update the Documentation</p>
                          <p className="text-muted">
                            Dwuamish Head, Seattle, WA 8:47 AM
                          </p>
                        </td>
                        <td className="td-actions text-right">
                          <Button
                            color="link"
                            id="tooltip170482171"
                            title=""
                            type="button"
                          >
                            <i className="tim-icons icon-settings" />
                          </Button>
                          <UncontrolledTooltip
                            delay={0}
                            target="tooltip170482171"
                          >
                            Edit Task
                          </UncontrolledTooltip>
                        </td>
                      </tr>
                      <tr>
                        <td>
                          <FormGroup check>
                            <Label check>
                              <Input
                                defaultChecked
                                defaultValue=""
                                type="checkbox"
                              />
                              <span className="form-check-sign">
                                <span className="check" />
                              </span>
                            </Label>
                          </FormGroup>
                        </td>
                        <td>
                          <p className="title">GDPR Compliance</p>
                          <p className="text-muted">
                            Alki Ave SW, Seattle, WA 98116, SUA 12:29 PM
                          </p>
                        </td>
                        <td className="td-actions text-right">
                          <Button
                            color="link"
                            id="tooltip720626938"
                            title=""
                            type="button"
                          >
                            <i className="tim-icons icon-settings" />
                          </Button>
                          <UncontrolledTooltip
                            delay={0}
                            target="tooltip720626938"
                          >
                            Edit Task
                          </UncontrolledTooltip>
                        </td>
                      </tr>
                      <tr>
                        <td>
                          <FormGroup check>
                            <Label check>
                              <Input defaultValue="" type="checkbox" />
                              <span className="form-check-sign">
                                <span className="check" />
                              </span>
                            </Label>
                          </FormGroup>
                        </td>
                        <td>
                          <p className="title">Export the processed files</p>
                          <p className="text-muted">
                            Capitol Hill, Seattle, WA 12:34 AM
                          </p>
                        </td>
                        <td className="td-actions text-right">
                          <Button
                            color="link"
                            id="tooltip598446371"
                            title=""
                            type="button"
                          >
                            <i className="tim-icons icon-settings" />
                          </Button>
                          <UncontrolledTooltip
                            delay={0}
                            target="tooltip598446371"
                          >
                            Edit Task
                          </UncontrolledTooltip>
                        </td>
                      </tr>
                      <tr>
                        <td>
                          <FormGroup check>
                            <Label check>
                              <Input defaultValue="" type="checkbox" />
                              <span className="form-check-sign">
                                <span className="check" />
                              </span>
                            </Label>
                          </FormGroup>
                        </td>
                        <td>
                          <p className="title">Release v2.0.0</p>
                          <p className="text-muted">
                            Ra Ave SW, Seattle, WA 98116, SUA 11:19 AM
                          </p>
                        </td>
                        <td className="td-actions text-right">
                          <Button
                            color="link"
                            id="tooltip797367748"
                            title=""
                            type="button"
                          >
                            <i className="tim-icons icon-settings" />
                          </Button>
                          <UncontrolledTooltip
                            delay={0}
                            target="tooltip797367748"
                          >
                            Edit Task
                          </UncontrolledTooltip>
                        </td>
                      </tr>
                      <tr>
                        <td>
                          <FormGroup check>
                            <Label check>
                              <Input defaultValue="" type="checkbox" />
                              <span className="form-check-sign">
                                <span className="check" />
                              </span>
                            </Label>
                          </FormGroup>
                        </td>
                        <td>
                          <p className="title">Solve the issues</p>
                          <p className="text-muted">
                            Caption Hill, LA 12:34 AM
                          </p>
                        </td>
                        <td className="td-actions text-right">
                          <Button
                            color="link"
                            id="tooltip147107903"
                            title=""
                            type="button"
                          >
                            <i className="tim-icons icon-settings" />
                          </Button>
                          <UncontrolledTooltip
                            delay={0}
                            target="tooltip147107903"
                          >
                            Edit Task
                          </UncontrolledTooltip>
                        </td>
                      </tr>
                      <tr>
                        <td>
                          <FormGroup check>
                            <Label check>
                              <Input defaultValue="" type="checkbox" />
                              <span className="form-check-sign">
                                <span className="check" />
                              </span>
                            </Label>
                          </FormGroup>
                        </td>
                        <td>
                          <p className="title">Arival at export process</p>
                          <p className="text-muted">
                            Capitol Hill, Seattle, WA 12:34 AM
                          </p>
                        </td>
                        <td className="td-actions text-right">
                          <Button
                            color="link"
                            id="tooltip841399405"
                            title=""
                            type="button"
                          >
                            <i className="tim-icons icon-settings" />
                          </Button>
                          <UncontrolledTooltip
                            delay={0}
                            target="tooltip841399405"
                          >
                            Edit Task
                          </UncontrolledTooltip>
                        </td>
                      </tr>
                    </tbody>
                  </Table>
                </div>
              </CardBody>
            </Card>
            <Card className="card-contributions">
              <CardBody>
                <CardTitle tag="h1">6,332</CardTitle>
                <h3 className="card-category">Total Public Contributions</h3>
                <p className="card-description">
                  After a very successful two-year run, we’re going to be
                  changing the way that contributions work.
                </p>
              </CardBody>
              <hr />
              <CardFooter>
                <Row>
                  <Col className="ml-auto mr-auto" lg="6" md="8">
                    <div className="card-stats justify-content-center">
                      <div className="d-flex justify-content-center align-items-center">
                        <span className="mr-2">Off</span>
                        <CustomInput
                          type="switch"
                          id="switch-1"
                          defaultChecked
                          className="mt-n4"
                        />
                        <span className="ml-n4">On</span>
                      </div>
                      <span>All public contributions</span>
                    </div>
                  </Col>
                  <Col className="ml-auto mr-auto" lg="6" md="8">
                    <div className="card-stats justify-content-center">
                      <div className="d-flex justify-content-center align-items-center">
                        <span className="mr-2">Off</span>
                        <CustomInput
                          type="switch"
                          id="switch-2"
                          className="mt-n4"
                        />
                        <span className="ml-n4">On</span>
                      </div>
                      <span>Past week contributions</span>
                    </div>
                  </Col>
                </Row>
              </CardFooter>
            </Card>
          </Col>
          <Col lg="6" sm="6">
            <Card className="card-timeline card-plain">
              <CardBody>
                <ul className="timeline timeline-simple">
                  <li className="timeline-inverted">
                    <div className="timeline-badge danger">
                      <i className="tim-icons icon-bag-16" />
                    </div>
                    <div className="timeline-panel">
                      <div className="timeline-heading">
                        <Badge color="danger">Some Title</Badge>
                      </div>
                      <div className="timeline-body">
                        <p>
                          Wifey made the best Father's Day meal ever. So
                          thankful so happy so blessed. Thank you for making my
                          family We just had fun with the “future” theme !!! It
                          was a fun night all together ... The always rude Kanye
                          Show at 2am Sold Out Famous viewing @ Figueroa and
                          12th in downtown.
                        </p>
                      </div>
                      <h6>
                        <i className="ti-time" />
                        11 hours ago via Twitter
                      </h6>
                    </div>
                  </li>
                  <li className="timeline-inverted">
                    <div className="timeline-badge success">
                      <i className="tim-icons icon-gift-2" />
                    </div>
                    <div className="timeline-panel">
                      <div className="timeline-heading">
                        <Badge color="success">Another One</Badge>
                      </div>
                      <div className="timeline-body">
                        <p>
                          Thank God for the support of my wife and real friends.
                          I also wanted to point out that it’s the first album
                          to go number 1 off of streaming!!! I love you Ellen
                          and also my number one design rule of anything I do
                          from shoes to music to homes is that Kim has to like
                          it....
                        </p>
                      </div>
                    </div>
                  </li>
                  <li className="timeline-inverted">
                    <div className="timeline-badge info">
                      <i className="tim-icons icon-planet" />
                    </div>
                    <div className="timeline-panel">
                      <div className="timeline-heading">
                        <Badge color="info">Another Title</Badge>
                      </div>
                      <div className="timeline-body">
                        <p>
                          Called I Miss the Old Kanye That’s all it was Kanye
                          And I love you like Kanye loves Kanye Famous viewing @
                          Figueroa and 12th in downtown LA 11:10PM
                        </p>
                        <p>
                          What if Kanye made a song about Kanye Royère doesn't
                          make a Polar bear bed but the Polar bear couch is my
                          favorite piece of furniture we own It wasn’t any
                          Kanyes Set on his goals Kanye
                        </p>
                        <hr />
                      </div>
                      <div className="timeline-footer">
                        <UncontrolledDropdown>
                          <DropdownToggle
                            caret
                            className="btn-round"
                            color="info"
                            data-toggle="dropdown"
                            type="button"
                          >
                            <i className="tim-icons icon-bullet-list-67" />
                          </DropdownToggle>
                          <DropdownMenu>
                            <DropdownItem
                              href="#pablo"
                              onClick={(e) => e.preventDefault()}
                            >
                              Action
                            </DropdownItem>
                            <DropdownItem
                              href="#pablo"
                              onClick={(e) => e.preventDefault()}
                            >
                              Another action
                            </DropdownItem>
                            <DropdownItem
                              href="#pablo"
                              onClick={(e) => e.preventDefault()}
                            >
                              Something else here
                            </DropdownItem>
                          </DropdownMenu>
                        </UncontrolledDropdown>
                      </div>
                    </div>
                  </li>
                </ul>
              </CardBody>
            </Card>
          </Col>
        </Row>
        <Row>
          <Col lg="3" md="6">
            <Card className="card-pricing card-primary card-white">
              <CardBody>
                <CardTitle tag="h1">pro</CardTitle>
                <CardImg
                  alt="..."
                  src={require("assets/img/card-primary.png").default}
                />
                <ListGroup>
                  <ListGroupItem>300 messages</ListGroupItem>
                  <ListGroupItem>150 emails</ListGroupItem>
                  <ListGroupItem>24/7 Support</ListGroupItem>
                </ListGroup>
                <div className="card-prices">
                  <h3 className="text-on-front">
                    <span>$</span>
                    95
                  </h3>
                  <h5 className="text-on-back">95</h5>
                  <p className="plan">Professional plan</p>
                </div>
              </CardBody>
              <CardFooter className="text-center mb-3 mt-3">
                <Button className="btn-round btn-just-icon" color="primary">
                  Get started
                </Button>
              </CardFooter>
            </Card>
          </Col>
          <Col lg="3" md="6">
            <Card className="card-pricing card-primary">
              <CardBody>
                <CardTitle tag="h1">pro</CardTitle>
                <CardImg
                  alt="..."
                  src={require("assets/img/card-primary.png").default}
                />
                <ListGroup>
                  <ListGroupItem>300 messages</ListGroupItem>
                  <ListGroupItem>150 emails</ListGroupItem>
                  <ListGroupItem>24/7 Support</ListGroupItem>
                </ListGroup>
                <div className="card-prices">
                  <h3 className="text-on-front">
                    <span>$</span>
                    95
                  </h3>
                  <h5 className="text-on-back">95</h5>
                  <p className="plan">Professional plan</p>
                </div>
              </CardBody>
              <CardFooter className="text-center mb-3 mt-3">
                <Button className="btn-round btn-just-icon" color="primary">
                  Get started
                </Button>
              </CardFooter>
            </Card>
          </Col>
          <Col md="6">
            <Card className="card-testimonial">
              <CardHeader className="card-header-avatar">
                <a href="#pablo" onClick={(e) => e.preventDefault()}>
                  <img
                    alt="..."
                    className="img img-raised"
                    src={require("assets/img/james.jpg").default}
                  />
                </a>
              </CardHeader>
              <CardBody>
                <p className="card-description">
                  The networking at Web Summit is like no other European tech
                  conference.
                </p>
                <div className="icon icon-primary">
                  <i className="fa fa-quote-right" />
                </div>
              </CardBody>
              <CardFooter>
                <CardTitle tag="h4">Robert Priscen</CardTitle>
                <p className="category">@robertpriscen</p>
              </CardFooter>
            </Card>
          </Col>
        </Row>
      </div>
    </>
  );
};

export default Widgets;
